src/nvme/types.h

```diff
@@ -9007,4 +9025,324 @@ struct nvme_ns_mgmt_host_sw_specified {
+ * state of a NVMe Controller's I/O Submission and I/O Completion queues
+ * @hdr: Header
+ * @sqs: I/O Submission Queue list
+ * @cqs: I/O Completion Queue list
+ */
+struct nvme_lm_nvme_controller_state_data {
+	struct nvme_lm_nvme_controller_state_data_header hdr;
+	union {
+		struct nvme_lm_io_submission_queue_data sqs[0];
+		struct nvme_lm_io_completion_queue_data cqs[0];
+	};
+};
+
+/**
+ * struct nvme_lm_controller_state_data_header - Controller State data header structure describes
+ * the contents of the Controller State data
+ * @ver:     Version of this data structure
+ * @csattr:  Controller state attributes
+ * @rsvd3:   Reserved
+ * @nvmecss: NVMe Controller state size in dwords
+ * @vss:     Vendor specific size in dowrds
+ */
+struct nvme_lm_controller_state_data_header {
+	__u16 ver;
+	__u8  csattr;
+	__u8  rsvd3[13];
+	__u8  nvmecss[16];
+	__u8  vss[16];
+};
+
+/**
+ * struct nvme_lm_controller_state_data - Controller State data structure contains data on the
+ * controller's state.
+ * @hdr:  Header
+ * @data: Data
+ */
+struct nvme_lm_controller_state_data {
+	struct nvme_lm_controller_state_data_header hdr;
+	struct nvme_lm_nvme_controller_state_data   data;
+};
+
+/**
+ * enum nvme_lm_queue_attributes - I/O Submission and I/O Completion Queue Attributes
+ * @NVME_LM_IOSQPC_MASK:	Mask to get the Physically Contiguous (PC) bit for this I/O
+ *				submission queue.
+ * @NVME_LM_IOSQPC_SHIFT:	Shift to get the PC bit for this I/O submission queue
+ * @NVME_LM_IOSQPRIO_MASK:	Mask to get the Priority for this I/O submission queue.
+ * @NVME_LM_IOSQPRIO_SHIFT:	Shift to get the Priority for this I/O submission queue.
+ * @NVME_LM_IOCQPC_MASK:	Mask to get the Physicaly Contiguous (PC) bit for this I/O
+ *				completion queue.
+ * @NVME_LM_IOCQPC_SHIFT:	Shift to get the PC bit for this I/O completion queue.
+ * @NVME_LM_IOCQIEN_MASK:	Mask to get the Interrupts Enabled bit for this I/O completion
+ *				queue
+ * @NVME_LM_IOCQIEN_SHIFT:	Shift to get the Interrupts Enabled bit for this I/O completion
+ * @NVME_LM_S0PT_MASK:		Mask to get the value of the Phase Tag bit for Slot 0 of this I/O
+ *				completion queue.
+ * @NVME_LM_S0PT_SHIFT:		Shift to get the value of the Phase Tag bit for Slot 0 of this I/O
+ *				completion queue.
+ * @NVME_LM_IOCQIV_MASK:	Mask to get the Interrupt Vector (IV) for this I/O completion
+ *				queue.
+ * @NVME_LM_IOCQIV_SHIFT:	Shift to get the IV for this I/O completion queue.
+ */
+enum nvme_lm_queue_attributes {
+	/* I/O Submission Queue */
+	NVME_LM_IOSQPC_MASK	= 0x1,
+	NVME_LM_IOSQPC_SHIFT	= 0,
+	NVME_LM_IOSQPRIO_MASK	= 0x3,
+	NVME_LM_IOSQPRIO_SHIFT	= 1,
+	/* I/O Completion Queue */
+	NVME_LM_IOCQPC_MASK	= 0x1,
+	NVME_LM_IOCQPC_SHIFT	= 0,
+	NVME_LM_IOCQIEN_MASK	= 0x1,
+	NVME_LM_IOCQIEN_SHIFT	= 1,
+	NVME_LM_S0PT_MASK	= 0x1,
+	NVME_LM_S0PT_SHIFT	= 2,
+	NVME_LM_IOCQIV_MASK	= 0xffff,
+	NVME_LM_IOCQIV_SHIFT	= 16,
```

Comment:
the same comment as above, with prefixing it consistently and matching the names with the spec.